src/modules/iotjs_module_websocket.c

```diff
@@ -243,8 +254,17 @@ JS_FUNCTION(PrepareHandshakeRequest) {
     return JS_CREATE_ERROR(COMMON, "Invalid host and/or path arguments!");
   };
 
-  generated_key = ws_generate_key(jsref);
-  size_t generated_key_len = strlen((char *)generated_key);
+  void *native_p;
+  JNativeInfoType *out_native_info;
+  bool has_p =
+      jerry_get_object_native_pointer(jsref, &native_p, &out_native_info);
+  if (!has_p || out_native_info != &wsclient_native_info) {
+    return WS_ERR_NATIVE_POINTER_ERR;
+  }
+  iotjs_wsclient_t *wsclient = (iotjs_wsclient_t *)native_p;
+
+  wsclient->generated_key = ws_generate_key(jsref);
```

Comment:
In the ws_generate_key, the base64 encode returns the buffer length, would it make sense to use directly that value? Maybe extend the `ws_generate_key` to accept an additional output parameter?

Reply:
Sure, good idea!